pandora-server-directory/src/database/validatedCollection.ts

```diff
@@ -192,3 +224,40 @@ export class ValidatedCollection<T extends Document> {
 		}
 	}
 }
+
+class ValidatedAsyncIter<T extends Document> implements AsyncIterableIterator<T | null> {
```

Comment:
[optional] This could be simplified by making use of async generator functions: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator